packages/story-editor/src/components/library/panes/shopping/product.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { Text, THEME_CONSTANTS } from '@googleforcreators/design-system';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../../../app/config';
+import ProductImage from './productImage';
+import ProductButton from './productButton';
+
+const StyledDescription = styled.div`
+  padding-left: 12px;
+`;
+
+const StyledTitle = styled(Text).attrs({
+  size: THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.MEDIUM,
+})`
+  color: ${({ theme }) => theme.colors.fg.primary};
+`;
+
+const StyledPrice = styled(Text)`
+  color: ${({ theme }) => theme.colors.fg.tertiary};
+`;
+function ProductPrice({ price, currency }) {
+  const {
+    locale: { locale },
+  } = useConfig();
+
+  if (price) {
+    return (
+      <StyledPrice>
+        {new Intl.NumberFormat(locale, {
+          style: 'currency',
+          currency: currency,
+        }).format(price)}
```

Comment:
This feels like it should a function that lives in a util.

Reply:
Thought about that but there's no `custom` code ...  we would only be test a built in function and seeing what happens when we pass good / bad values to it (which the schema should pick up).

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat

Reply:
Added some extra guards for the variables 

https://github.com/GoogleForCreators/web-stories-wp/pull/11392/files#diff-78c8fc16b6c6f39ae876dbd1d30b2674df52e81874c83a53af3c559052a57e7dR39

Can break it out as need be in the future.